Doc/library/typing.rst

```diff
@@ -1769,21 +1769,11 @@ These are not used in annotations. They are building blocks for declaring types.
       assert Point2D(x=1, y=2, label='first') == dict(x=1, y=2, label='first')
 
    To allow using this feature with older versions of Python that do not
```

Comment:
"versions that do not support PEP 526" means 3.5 or lower, which we no longer care about. We should rewrite this text to motivate the feature only with case for keys that are not valid identifiers (below) and omit mention of PEP 526.

Reply:
Like this?

```
>>> TD = TypedDict("TD", iter([("a", int)]))
>>> TD.__annotations__
{'a': <class 'int'>}
```

We shouldn't mention that as type checkers won't support it.

Reply:
Yup. The runtime is very duck-type-y, but PEP 589 specifically says:

> A type checker is only expected to accept a dictionary display expression as the second argument to TypedDict

https://peps.python.org/pep-0589/#alternative-syntax

So we shouldn't advertise that you can use other iterables for the second second argument :)